First pass consenting prototype

matthewmascord opened the pull request:
First pass at a consenting page based on GDS using Nunjucks.

matthewmascord committed:
First pass consenting prototype

nataliecarey commented on app/assets/sass/components/_cookie-banner-new.scss:
```diff
@@ -0,0 +1,31 @@
+.app-cookie-banner-new__button {
+  width: 100%;
+
+  @media (min-width: 48.0625em) {
```
You should be able to use

```
@include govuk-media-query($from: tablet) { }
```

See example here https://github.com/hmrc/hmrc-frontend/blob/df356955199a820a6ec4cb761956b8f275e6da16/src/components/status-tags-in-task-list-pages/_status-tags-in-task-list-pages.scss#L13

nataliecarey commented on lib/extensions/_extensions.scss:
```diff
@@ -0,0 +1 @@
+$govuk-extensions-url-context: "/extension-assets"; @import "/Users/matthewmascord/projects/hmrc/tracking-consent-prototype/node_modules/govuk-frontend/govuk/all.scss";
```
Fully qualified path used, it should be relative.

nataliecarey replied:
Looking deeper at it - this should be git ignored as it's a generated file.  For now please add it to `.gitignore` and I'll contact GDS.

nataliecarey commented on lib/utils.js:
```diff
@@ -291,6 +291,9 @@ exports.autoStoreData = function (req, res, next) {
+    // do not show cookie banner on cookie consent page
+    app.locals.shouldShowNewCookieMessage = req.path !== '/cookie-consent'
+
     const COOKIE_NAME = 'seen_cookie_message'
```
As the purpose of this prototype is to demonstrate the look and feel of the banner I'd suggest we change this condition.

I'd set up a page identical to index.html except that it doesn't show the cookie banner.

nataliecarey commented on lib/utils.js:
```diff
@@ -291,6 +291,9 @@ exports.autoStoreData = function (req, res, next) {
 exports.handleCookies = function (app) {
   return function handleCookies (req, res, next) {
+    // do not show cookie banner on cookie consent page
+    app.locals.shouldShowNewCookieMessage = req.path !== '/cookie-consent'
```
I think this is a scoping issue as `app.locals` is essentially a global object.  There's a theoretical race condition here.

As this is a demo of the banner I wonder whether we should just set this variable in the views themselves, or set up routes which render with a model.

matthewmascord committed:
Fix PR issues

Remove and add _extensions.scss to gitignore
Reverse logic of showing cookie banner and set in view
Use proper gov media selector
